onnxruntime/core/framework/utils.h

```diff
@@ -26,6 +23,11 @@ class IExecutionProvider;
 class Node;
 class Tensor;
 struct KernelCreateInfo;
+#ifdef ENABLE_TRAINING
+struct PartialGraphExecutionState;
+typedef InlinedHashMap<std::string, OrtValue> OrtValueCache;
+typedef std::shared_ptr<OrtValueCache> OrtValueCachePtr;
```

Comment:
> std::shared_ptr<OrtValueCache [](http://example.com/codeflow?start=7&length=30)

Same as before, do we need to share OrtValues with the shared_ptr?

Reply:
save as the comments above.